src/dfi/validation.cpp

```diff
@@ -3416,10 +3420,11 @@ static Res ForceCloseAllLoans(const CBlockIndex *pindex, CCustomCSView &cache, B
             for (const auto &loan : loanAmounts->balances) {
                 LogPrintf("    %s@%d\n", GetDecimalString(loan.second), loan.first.v);
             }
-            const auto collAmounts = cache.GetVaultCollaterals(vaultId);
-            LogPrintf("%d collaterals: \n", collAmounts->balances.size());
-            for (const auto &loan : collAmounts->balances) {
-                LogPrintf("    %s@%d\n", GetDecimalString(loan.second), loan.first.v);
+            if (const auto collAmounts = cache.GetVaultCollaterals(vaultId)) {
+                LogPrintf("%d collaterals: \n", collAmounts->balances.size());
+                for (const auto &loan : collAmounts->balances) {
+                    LogPrintf("    %s@%d\n", GetDecimalString(loan.second), loan.first.v);
+                }
```

Comment:
I suggest adding a log here in case the collAmounts are empty. Then we at least have the info why it failed.